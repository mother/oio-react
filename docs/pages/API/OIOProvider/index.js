import React from 'react'
import APITableBlock from '../../../components/APITableBlock'
import HeaderBlock from '../../../components/HeaderBlock'
import SpecimenBlock from '../../../components/SpecimenBlock'
import OIOProviderSpecimen from '../../../specimens/OIOProviderComponent'
import oioSpec from '../../../../src/OIOProvider/spec'

const OIOProviderPage = () => (
   <div>
      <HeaderBlock title="OIOProvider" />
      <SpecimenBlock specimen={OIOProviderSpecimen} />
      <APITableBlock spec={oioSpec} title="Component API" />
   </div>
)

export default OIOProviderPage
