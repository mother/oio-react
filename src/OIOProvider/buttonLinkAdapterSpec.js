export default [{
   name: 'isActive',
   type: 'Function',
   default: 'See Description',
   description: 'A function that is passed in a link. This function must return a boolean that represents whether the link is currently active. The default function does a simple check based on <code>window.location.pathname</code>. This is only by <code>ListMenuButton</code> and <code>TabMenuButton</code>, but not <code>Button</code>.'
}, {
   name: 'render',
   type: 'Function',
   default: 'See Description',
   description: 'A function that is passed the props passed in to the <code>Button</code> or <code>ListMenuButton</code> element. This function must return a React element. The default function simply returns <code>&lt;a href={linkTo}&gt;{children}&lt;/a&gt;</code>.'
}]
