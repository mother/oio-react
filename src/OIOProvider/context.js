import React from 'react'

const OIOContext = React.createContext()

const withOIOContext = WrappedComponent => React.forwardRef((props, ref) => (
   <OIOContext.Consumer>
      {context => (
         <WrappedComponent {...props} OIOContext={context} ref={ref} />
      )}
   </OIOContext.Consumer>
))

export { OIOContext as default, withOIOContext }
