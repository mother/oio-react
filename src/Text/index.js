import React, { Component } from 'react'
import { css, cx } from 'emotion'
import PropTypes from 'prop-types'
import generateStyleObject from '../utils/generators'

@generateStyleObject(({
   autoScale,
   baseFontSize,
   baseAutoScaleFontSize,
   color,
   fontFamily,
   scaleRatio,
   size,
   uppercase,
   weight
}) => {
   // Dynamically calculate font size
   const floatScaleRatio = parseFloat(scaleRatio)
   const adjustedTextSize = size < 1 ? floatScaleRatio * size : floatScaleRatio ** size
   let textSizeCoefficient = autoScale ? baseAutoScaleFontSize : baseFontSize
   textSizeCoefficient = uppercase ? textSizeCoefficient *= 0.9 : textSizeCoefficient

   return ({
      color,
      fontFamily,
      fontSize: `${textSizeCoefficient * adjustedTextSize}${autoScale ? 'vw' : 'px'}`,
      fontWeight: weight,
      lineHeight: size > 5 ? '120%' : '130%',
      textTransform: uppercase && 'uppercase'
   })
})

export default class Text extends Component {
   /* eslint-disable */
   static propTypes = {
      autoScale: PropTypes.bool,
      baseFontSize: PropTypes.string,
      baseAutoScaleFontSize: PropTypes.string,
      calculatedStyleProps: PropTypes.object.isRequired,
      children: PropTypes.node,
      className: PropTypes.string,
      color: PropTypes.string,
      fontFamily: PropTypes.string,
      scaleRatio: PropTypes.string,
      size: PropTypes.string,
      style: PropTypes.object,
      uppercase: PropTypes.bool,
      weight: PropTypes.string
   }
   /* eslint-enable */

   static defaultProps = {
      autoScale: false,
      baseFontSize: '16',
      baseAutoScaleFontSize: '4[a] 2.5[b] 1.75[c] 1.25[d] 1[e] 0.9[f]',
      scaleRatio: '1.125',
      size: '1',
      style: {},
      uppercase: false,
      weight: 'normal'
   }

   render() {
      const styles = {
         ...this.props.style,
         ...this.props.calculatedStyleProps
      }

      return (
         <div className={cx(css(styles), this.props.className)}>
            {this.props.children}
         </div>
      )
   }
}
