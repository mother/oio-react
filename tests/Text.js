import React from 'react'
import serializer from 'jest-emotion'
import { render } from 'react-testing-library'
import OIOProvider from '../src/OIOProvider'
import Text from '../src/Text'

// To show emotion styles in snapshot teset
expect.addSnapshotSerializer(serializer)

test('Text will render at different sizes correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text size="1">Size 1</Text>
         <Text size="2">Size 2</Text>
         <Text size="3">Size 3</Text>
         <Text size="4">Size 4</Text>
         <Text size="5">Size 5</Text>
         <Text size="1[a] 2[b] 3[c] 4[d] 5[e] 6[f]">
            Responsive Sizes
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render at autoScale sizes correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text size="1" autoScale>AutoScaled Size 1</Text>
         <Text size="2" autoScale>AutoScaled Size 2</Text>
         <Text size="3" autoScale>AutoScaled Size 3</Text>
         <Text size="4" autoScale>AutoScaled Size 4</Text>
         <Text size="5" autoScale>AutoScaled Size 5</Text>
         <Text size="1[a] 2[b] 3[c] 4[d] 5[e] 6[f]" autoScale>
            AutoScaled Responsive Sizes
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render at different font weights correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text weight="bold">Bold</Text>
         <Text weight="semibold">Semibold</Text>
         <Text weight="medium">Medium</Text>
         <Text weight="normal">Normal</Text>
         <Text weight="light">Light</Text>
         <Text weight="fakeWeight">Non Existing Font Weight</Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render color correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text color="red">Red Text</Text>
         <Text color="#ff0000">Red Text</Text>
         <Text color="rgba(255,0,0,1)">Red Text</Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render uppercase prop correctly', () => {
   const { container } = render(
      <OIOProvider>
         <Text uppercase>Uppercase Text</Text>
         <Text>Standard Case Text</Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})

test('Text will render incorrect prop values safely', () => {
   const { container } = render(
      <OIOProvider>
         <Text
            size="The Wrong Size"
            weight="fakeWeight">
            Non Existing Font Weight
         </Text>
      </OIOProvider>
   )

   expect(container).toMatchSnapshot()
})
