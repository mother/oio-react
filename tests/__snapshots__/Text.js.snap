// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text will render at autoScale sizes correctly 1`] = `
.emotion-6 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 2.8125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 2.025vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 1.6875vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 1.125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 0.84375vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 0.703125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-1 {
    font-size: 3.1640625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-1 {
    font-size: 2.278125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-1 {
    font-size: 1.8984375vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-1 {
    font-size: 1.265625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-1 {
    font-size: 0.94921875vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-1 {
    font-size: 0.791015625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-2 {
    font-size: 3.5595703125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-2 {
    font-size: 2.562890625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-2 {
    font-size: 2.1357421875vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-2 {
    font-size: 1.423828125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-2 {
    font-size: 1.06787109375vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-2 {
    font-size: 0.889892578125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-3 {
    font-size: 4.0045166015625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-3 {
    font-size: 2.8832519531250003vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-3 {
    font-size: 2.4027099609375vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-3 {
    font-size: 1.601806640625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-3 {
    font-size: 1.20135498046875vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-3 {
    font-size: 1.001129150390625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-4 {
    font-size: 4.5050811767578125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-4 {
    font-size: 3.243658447265625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-4 {
    font-size: 2.7030487060546875vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-4 {
    font-size: 1.802032470703125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-4 {
    font-size: 1.3515243530273438vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-4 {
    font-size: 1.1262702941894531vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-5 {
    font-size: 2.8125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-5 {
    font-size: 2.278125vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-5 {
    font-size: 2.1357421875vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-5 {
    font-size: 1.601806640625vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-5 {
    font-size: 1.3515243530273438vw;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-5 {
    font-size: 1.2670540809631348vw;
    font-weight: 400;
    line-height: 110%;
  }
}

<div>
  <div
    class=" emotion-6"
  >
    <div
      class=" emotion-0"
    >
      AutoScaled Size 1
    </div>
    <div
      class=" emotion-1"
    >
      AutoScaled Size 2
    </div>
    <div
      class=" emotion-2"
    >
      AutoScaled Size 3
    </div>
    <div
      class=" emotion-3"
    >
      AutoScaled Size 4
    </div>
    <div
      class=" emotion-4"
    >
      AutoScaled Size 5
    </div>
    <div
      class=" emotion-5"
    >
      AutoScaled Responsive Sizes
    </div>
  </div>
</div>
`;

exports[`Text will render at different font weights correctly 1`] = `
.emotion-6 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-3 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-3 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-3 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-3 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-3 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-3 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 15.662109375px;
    font-weight: 700;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 600;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 500;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-4 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-4 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-4 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-4 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-4 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-4 {
    font-size: 15.662109375px;
    font-weight: 300;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-5 {
    font-size: 15.662109375px;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-5 {
    font-size: 15.662109375px;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-5 {
    font-size: 15.662109375px;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-5 {
    font-size: 15.662109375px;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-5 {
    font-size: 15.662109375px;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-5 {
    font-size: 15.662109375px;
    line-height: 120%;
  }
}

<div>
  <div
    class=" emotion-6"
  >
    <div
      class=" emotion-0"
    >
      Bold
    </div>
    <div
      class=" emotion-1"
    >
      Semibold
    </div>
    <div
      class=" emotion-2"
    >
      Medium
    </div>
    <div
      class=" emotion-3"
    >
      Normal
    </div>
    <div
      class=" emotion-4"
    >
      Light
    </div>
    <div
      class=" emotion-5"
    >
      Non Existing Font Weight
    </div>
  </div>
</div>
`;

exports[`Text will render at different sizes correctly 1`] = `
.emotion-6 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-1 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-1 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-1 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-1 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-1 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-1 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-2 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-3 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-3 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-3 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-3 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-3 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-3 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-4 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-4 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-4 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-4 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-4 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-4 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-5 {
    font-size: 12.375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-5 {
    font-size: 13.921875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-5 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-5 {
    font-size: 17.619873046875px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-5 {
    font-size: 19.822357177734375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-5 {
    font-size: 22.300151824951172px;
    font-weight: 400;
    line-height: 110%;
  }
}

<div>
  <div
    class=" emotion-6"
  >
    <div
      class=" emotion-0"
    >
      Size 1
    </div>
    <div
      class=" emotion-1"
    >
      Size 2
    </div>
    <div
      class=" emotion-2"
    >
      Size 3
    </div>
    <div
      class=" emotion-3"
    >
      Size 4
    </div>
    <div
      class=" emotion-4"
    >
      Size 5
    </div>
    <div
      class=" emotion-5"
    >
      Responsive Sizes
    </div>
  </div>
</div>
`;

exports[`Text will render color correctly 1`] = `
.emotion-3 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    color: red;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-1 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-1 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-1 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-1 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-1 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-1 {
    color: #ff0000;
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-2 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-2 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-2 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-2 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-2 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-2 {
    color: rgba(255,0,0,1);
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

<div>
  <div
    class=" emotion-3"
  >
    <div
      class=" emotion-0"
    >
      Red Text
    </div>
    <div
      class=" emotion-1"
    >
      Red Text
    </div>
    <div
      class=" emotion-2"
    >
      Red Text
    </div>
  </div>
</div>
`;

exports[`Text will render incorrect prop values safely 1`] = `
.emotion-1 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    line-height: 120%;
  }
}

<div>
  <div
    class=" emotion-1"
  >
    <div
      class=" emotion-0"
    >
      Non Existing Font Weight
    </div>
  </div>
</div>
`;

exports[`Text will render uppercase prop correctly 1`] = `
.emotion-2 {
  font-family: sans-serif;
  font-size: 16px;
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:1700px) {
  .emotion-1 {
    font-size: 15.662109375px;
    font-weight: 400;
    line-height: 120%;
  }
}

@media (min-width:0px) and (max-width:475px) {
  .emotion-0 {
    font-size: 14.0958984375px;
    font-weight: 400;
    line-height: 120%;
    text-transform: uppercase;
  }
}

@media (min-width:475px) and (max-width:675px) {
  .emotion-0 {
    font-size: 14.0958984375px;
    font-weight: 400;
    line-height: 120%;
    text-transform: uppercase;
  }
}

@media (min-width:675px) and (max-width:1000px) {
  .emotion-0 {
    font-size: 14.0958984375px;
    font-weight: 400;
    line-height: 120%;
    text-transform: uppercase;
  }
}

@media (min-width:1000px) and (max-width:1350px) {
  .emotion-0 {
    font-size: 14.0958984375px;
    font-weight: 400;
    line-height: 120%;
    text-transform: uppercase;
  }
}

@media (min-width:1350px) and (max-width:1700px) {
  .emotion-0 {
    font-size: 14.0958984375px;
    font-weight: 400;
    line-height: 120%;
    text-transform: uppercase;
  }
}

@media (min-width:1700px) {
  .emotion-0 {
    font-size: 14.0958984375px;
    font-weight: 400;
    line-height: 120%;
    text-transform: uppercase;
  }
}

<div>
  <div
    class=" emotion-2"
  >
    <div
      class=" emotion-0"
    >
      Uppercase Text
    </div>
    <div
      class=" emotion-1"
    >
      Standard Case Text
    </div>
  </div>
</div>
`;
